distributed/tests/test_worker.py

```diff
@@ -899,28 +900,81 @@ def __sizeof__(self):
     assert result.data == 123
 
 
+class Bad:
+    def __init__(self, size):
+        self.size = size
+
+    def __getstate__(self):
+        raise TypeError()
+
+    def __sizeof__(self):
+        return self.size
+
+
 @gen_cluster(client=True)
 async def test_fail_write_to_disk(c, s, a, b):
-    class Bad:
-        def __getstate__(self):
-            raise TypeError()
 
-        def __sizeof__(self):
-            return int(100e9)
-
-    future = c.submit(Bad)
+    # Key immediately stored to disk, fails to serialize
+    future = c.submit(Bad, int(100e9))
     await wait(future)
 
     assert future.status == "error"
 
-    with pytest.raises(TypeError):
+    with pytest.raises(TypeError, match="Could not serialize"):
         await future
 
     futures = c.map(inc, range(10))
     results = await c._gather(futures)
     assert results == list(map(inc, range(10)))
 
 
+try:
+    import zict
+except ImportError:
+    zict = None
+requires_zict_210 = pytest.mark.skipif(
+    not zict or parse_version(zict.__version__) <= parse_version("2.0.0"),
+    reason="requires zict version > 2.0.0",
+)
+
+
+@requires_zict_210
+@gen_cluster(
+    client=True,
+    nthreads=[("", 1)],
+    worker_kwargs=dict(
+        memory_limit=2000,
+        memory_target_fraction=False,
+        memory_spill_fraction=0.5,
+        memory_pause_fraction=False,
+    ),
+)
+async def test_fail_write_to_disk_evict(c, s, w):
+    """Test eviction triggered due to spill fraction and no target memory."""
+    np = pytest.importorskip("numpy")
+    # size on memory 500
+    bad = c.submit(Bad, 500, key="bad")
+    await wait(bad)
+
+    # key is in fast
+    assert bad.status == "finished"
+
+    x = c.submit(np.zeros, 500, dtype="u1", key="x")
+
+    with captured_logger(logging.getLogger("distributed.spill")) as logs_evict_key:
+        await wait(x)
+
+    # tries to evict bad key, fails to serialize, key remains in fast
+    # This test is not passing since this is empty. I think we should have use the logs
+    # from logging.getLogger("distributed.protocol.pickle")
+    logs_value = logs_evict_key.getvalue()
+    assert "Failed to pickle" in logs_value
+    assert "Traceback" in logs_value
+    assert set(w.data) == {x.key, bad.key}
+    assert set(w.data.memory) == {x.key, bad.key}
+    assert not w.data.disk
```

Comment:
@crusaderky this test `test_worker.py::test_fail_write_to_disk_evict`fails because the logs from `"distirbuted.spill"` are empty. Since we are in the worker, don't we need to us the logs from `"distributed.protocol.pickle"` ?

Reply:
Why are they empty? Why are we not hitting the logging mechanism in SpillBuffer?